### PR TITLE
PCA Scenarios for review

### PR DESCRIPTION
**Pull request by vpalazhi**
These are the PCA Labs which are currently running on the KodeKloud Account. 

Please review and let me know should you want to make any changes. 

For now, the labs on the course will still use the kodekloud account until the review is complete.

### CODE REVIEW

**Comment by wallnerryan** [comment]
Thanks @vpalazhi I'll review this tomorrow.

**Review by wallnerryan — COMMENTED** [review]
i have not forgot about this @vpalazhi just taking a bit to get through the review. Rest assures, its in progress.

**Review by wallnerryan — APPROVED** [review]

File: portworx-3dsnapshots/assets/create-vols.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+ssh -o strictHostKeyChecking=no node01 "pxctl volume create testvol"
+
+ssh -o strictHostKeyChecking=no node02 "pxctl volume create local_volume --nodes=LocalNode"
+
+ssh -o strictHostKeyChecking=no node03 "pxctl volume create ha_volume --nodes=LocalNode --repl3"
```
[the remark this answers is not in the document]
Comment (wallnerryan) [reply portworx-3dsnapshots/assets/create-vols.sh:5]: will fix after merge.

File: portworx-application-backup/assets/create-objectstores.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+MINIO_ENDPOINT2=$(kubectl describe svc px-minio-2 | grep Endpoints | awk '{print $2}')
+
+mc config host add px http://$MINIO_ENDPOINT myaccesskey mysecretkey S3v4
+
+mc config host add object-store-2 http://$MINIO_ENDPOINT2 AAXXBBCCDD Sup3rS3cret S3v4
+
+
+ssh node01 "/opt/pwx/bin/pxctl cred create   --provider s3   \
+    --s3-access-key AABBCCDDEEFF   --s3-secret-key N3wSup3rS3cret   \
+    --s3-region us-east-1 --s3-endpoint http://10.34.0.3:9000 objectstore-2"
```
Comment (wallnerryan) [inline portworx-application-backup/assets/create-objectstores.sh:18]: same here

File: portworx-application-backup/create-objectstores.sh [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+MINIO_ENDPOINT2=$(kubectl describe svc px-minio-2 | grep Endpoints | awk '{print $2}')
+
+mc config host add px http://$MINIO_ENDPOINT myaccesskey mysecretkey S3v4
+
+mc config host add object-store-2 http://$MINIO_ENDPOINT2 AAXXBBCCDD Sup3rS3cret S3v4
+
+
+ssh node01 "/opt/pwx/bin/pxctl cred create   --provider s3   \
+    --s3-access-key AABBCCDDEEFF   --s3-secret-key N3wSup3rS3cret   \
+    --s3-region us-east-1 --s3-endpoint http://10.34.0.3:9000 objectstore-2"
```
Comment (wallnerryan) [inline portworx-application-backup/create-objectstores.sh:18]: here too

File: portworx-cloud-credentails/assets/create-objectstores.sh [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+MINIO_ENDPOINT2=$(kubectl describe svc px-minio-2 | grep Endpoints | awk '{print $2}')
+
+mc config host add px http://$MINIO_ENDPOINT myaccesskey mysecretkey S3v4
+
+mc config host add object-store-2 http://$MINIO_ENDPOINT2 AAXXBBCCDD Sup3rS3cret S3v4
+
+
+ssh node01 "/opt/pwx/bin/pxctl cred create   --provider s3   \
+    --s3-access-key AABBCCDDEEFF   --s3-secret-key N3wSup3rS3cret   \
+    --s3-region us-east-1 --s3-endpoint http://10.34.0.3:9000 objectstore-2"
```
Comment (wallnerryan) [inline portworx-cloud-credentails/assets/create-objectstores.sh:18]: here to

File: portworx-cloud-credentails/create-objectstores.sh [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+MINIO_ENDPOINT2=$(kubectl describe svc px-minio-2 | grep Endpoints | awk '{print $2}')
+
+mc config host add px http://$MINIO_ENDPOINT myaccesskey mysecretkey S3v4
+
+mc config host add object-store-2 http://$MINIO_ENDPOINT2 AAXXBBCCDD Sup3rS3cret S3v4
+
+
+ssh node01 "/opt/pwx/bin/pxctl cred create   --provider s3   \
+    --s3-access-key AABBCCDDEEFF   --s3-secret-key N3wSup3rS3cret   \
+    --s3-region us-east-1 --s3-endpoint http://10.34.0.3:9000 objectstore-2"
```
Comment (wallnerryan) [inline portworx-cloud-credentails/create-objectstores.sh:18]: here too

File: portworx-cloud-snapshots/assets/create-objectstores.sh [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+MINIO_ENDPOINT2=$(kubectl describe svc px-minio-2 | grep Endpoints | awk '{print $2}')
+
+mc config host add px http://$MINIO_ENDPOINT myaccesskey mysecretkey S3v4
+
+mc config host add object-store-2 http://$MINIO_ENDPOINT2 AAXXBBCCDD Sup3rS3cret S3v4
+
+
+ssh node01 "/opt/pwx/bin/pxctl cred create   --provider s3   \
+    --s3-access-key AABBCCDDEEFF   --s3-secret-key N3wSup3rS3cret   \
+    --s3-region us-east-1 --s3-endpoint http://10.34.0.3:9000 objectstore-2"
```
Comment (wallnerryan) [inline portworx-cloud-snapshots/assets/create-objectstores.sh:18]: ""

File: portworx-cloud-snapshots/create-objectstores.sh [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+MINIO_ENDPOINT2=$(kubectl describe svc px-minio-2 | grep Endpoints | awk '{print $2}')
+
+mc config host add px http://$MINIO_ENDPOINT myaccesskey mysecretkey S3v4
+
+mc config host add object-store-2 http://$MINIO_ENDPOINT2 AAXXBBCCDD Sup3rS3cret S3v4
+
+
+ssh node01 "/opt/pwx/bin/pxctl cred create   --provider s3   \
+    --s3-access-key AABBCCDDEEFF   --s3-secret-key N3wSup3rS3cret   \
+    --s3-region us-east-1 --s3-endpoint http://10.34.0.3:9000 objectstore-2"
```
Comment (wallnerryan) [inline portworx-cloud-snapshots/create-objectstores.sh:18]: """

File: portworx-cloud-snapshots/step3.md [hunk 8]
```diff
@@ -0,0 +1,3 @@
+</br>
+
+##### We have deployed a mysql pod that uses portworx volume. Take a cloud snapshot of this PVC called `mysql-snapshot`. The snapshot should be successfully backed up to the object store.
```
Comment (wallnerryan) [inline portworx-cloud-snapshots/step3.md:3]: portworx --> Portworx

File: portworx-encryption/assets/create-vols.sh [hunk 9]
```diff
@@ -0,0 +1,6 @@
+ssh -o strictHostKeyChecking=no node01 "pxctl volume create testvol"
+
+ssh -o strictHostKeyChecking=no node02 "pxctl volume create local_volume --nodes=LocalNode"
+
+ssh -o strictHostKeyChecking=no node03 "pxctl volume create ha_volume --nodes=LocalNode --repl3"
```
Comment (wallnerryan) [inline portworx-encryption/assets/create-vols.sh:5]: --repl3 --> --repl 3

File: portworx-k8-sched-snap/assets/create-vols.sh [hunk 10]
```diff
@@ -0,0 +1,6 @@
+ssh -o strictHostKeyChecking=no node01 "pxctl volume create testvol"
+
+ssh -o strictHostKeyChecking=no node02 "pxctl volume create local_volume --nodes=LocalNode"
+
+ssh -o strictHostKeyChecking=no node03 "pxctl volume create ha_volume --nodes=LocalNode --repl3"
```
Comment (wallnerryan) [inline portworx-k8-sched-snap/assets/create-vols.sh:5]: need space or `=` in --repl3

File: portworx-k8-snapshots/index.json [hunk 11]
```diff
@@ -0,0 +1,72 @@
+{
+  "title": "Portworx Kubernetes Snapshots Practice Lab",
+  "description": "In this lab, we will get hands on with the creating volume snapshots and  restore operations in Kubernetes ",
```
Comment (wallnerryan) [inline portworx-k8-snapshots/index.json:3]: there is extra space in between and & restore `volume snapshots and  restore`

File: portworx-recap/step0.md [hunk 12]
```diff
@@ -0,0 +1,16 @@
+<p>In this lab, we will revisit some of the basic concepts we dsicussed in the **Portworx Fundamentals course**.</p>
+
+<p>This lab environment consists of a 3 node cluster. **Portworx** has already been installed and needs no additional configuration from the user.</p>
+
+<p> Going forward we will make use similar set up with a 3 node portworx cluster for all Labs. The labs will test you on the lessons learned in the associated lecture.</p>
+
+#### Lab Question Formats:
+
+###### Multiple Choice (Single Correct Answer)
+A circular check box preceeding the options imples a single correct answer.
```
Comment (wallnerryan) [inline portworx-recap/step0.md:10]: `imples` supposed to be "implies"?

File: portworx-resize-pvc/index.json [hunk 13]
```diff
@@ -0,0 +1,61 @@
+{
+  "title": "Portworx Kubernetes PVC Resize Practice Lab",
+  "description": "Hand-on lab to learn how to resize PVC",
```
Comment (wallnerryan) [inline portworx-resize-pvc/index.json:3]: `Hand-on` --> Hands-on 

PVC --> PVCs or `a PVC`

File: portworx-resize-pvc/step5.md [hunk 14]
```diff
@@ -0,0 +1 @@
+##### Q5: Create a group snapshot called 'group-nginx-snap' for the PVC's of the cassandra StatefulSet. Make sure that the snapshot can be restored to a new namespace called 'nginx-clone'.
```
Comment (wallnerryan) [inline portworx-resize-pvc/step5.md:1]: This is confusing? a group snapshot of nginx for cassandra ss? Is this just a copy paste issue?

File: portworx-security-roles-contexts/assets/create-vols.sh [hunk 15]
```diff
@@ -0,0 +1,6 @@
+ssh -o strictHostKeyChecking=no node01 "pxctl volume create testvol"
+
+ssh -o strictHostKeyChecking=no node02 "pxctl volume create local_volume --nodes=LocalNode"
+
+ssh -o strictHostKeyChecking=no node03 "pxctl volume create ha_volume --nodes=LocalNode --repl3"
```
Comment (wallnerryan) [inline portworx-security-roles-contexts/assets/create-vols.sh:5]: --repl3 need space or =

File: portworx-snapshots/assets/create-vols.sh [hunk 16]
```diff
@@ -0,0 +1,6 @@
+ssh -o strictHostKeyChecking=no node01 "pxctl volume create testvol"
+
+ssh -o strictHostKeyChecking=no node02 "pxctl volume create local_volume --nodes=LocalNode"
+
+ssh -o strictHostKeyChecking=no node03 "pxctl volume create ha_volume --nodes=LocalNode --repl3"
```
Comment (wallnerryan) [inline portworx-snapshots/assets/create-vols.sh:5]: repl3 --> repl 3 or repl=3

File: portworx-troubleshooting/step9.md [hunk 17]
```diff
@@ -0,0 +1,12 @@
+</br>
+
+##### The StorageClass `px-gamma-sc` does not have enough replicas to use a `db io profile`. Let us fix that. To do this, delete and recreate the `px-gamma-sc` storage class with the correct parameters:
```
Comment (wallnerryan) [inline portworx-troubleshooting/step9.md:3]: `db io profile`. -->  `db io_profile`.

**Comment by wallnerryan** [comment]
ping @vpalazhi for updates to the above comments before we can merge?

File: portworx-3dsnapshots/assets/create-mysql.yaml [hunk 18]
```diff
@@ -0,0 +1,74 @@
+---
+kind: StorageClass
+apiVersion: storage.k8s.io/v1beta1
+metadata:
+    name: px-ha-sc
+provisioner: kubernetes.io/portworx-volume
+parameters:
+   repl: "3"
+   io_profile: "db"
```
[the remark this answers is not in the document]
Comment (wallnerryan) [reply portworx-3dsnapshots/assets/create-mysql.yaml:9]: (no action needed)

**Comment by vpalazhi** [comment]
Hi @wallnerryan , sorry been busy. Wrapping up a course this week.

I will look in to this in the weekend.

I believe some of the files to be updated here are not used within the scenario but I will check and confirm.

**Comment by wallnerryan** [comment]
Thanks @vpalazhi yes please just verify these, most are repetitive, the `repl3` one definitely need to be fixed i believe but either way once these are validated i'll merge. Thanks.

**Comment by wallnerryan** [comment]
ping @vpalazhi

File: portworx-3dsnapshots/assets/create-vols.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (vpalazhi) [reply portworx-3dsnapshots/assets/create-vols.sh:5]: so this one - its set to --repl=3 which I think works:
node01 $ pxctl volume create ha_volume --nodes=LocalNode --repl=3
Volume successfully created: 230996163515780
node01 $

However, this was just a template script which is NOT used by this scenario . Validated the entire scenario and works well as intended.

File: portworx-application-backup/assets/create-objectstores.sh (hunk 2 above)
Comment (vpalazhi) [reply portworx-application-backup/assets/create-objectstores.sh:18]: This script is a template script and it's not actually used by the scenario at all. The mini host is set up by the load-quiz.sh script. 
Here we are making use of the endpoint from the service:



ssh -o strictHostKeyChecking=no node01 /opt/pwx/bin/pxctl cred create --provider s3 --s3-access-key ZZYYXXWWVVUU --s3-secret-key 0ldSup3rS3cr3t --s3-region us-west-2 --s3-endpoint http://$MINIO_ENDPOINT px-s3-store

File: portworx-application-backup/create-objectstores.sh (hunk 3 above)
Comment (vpalazhi) [reply portworx-application-backup/create-objectstores.sh:18]: This script is a template script and it's not actually used by the scenario at all. The mini host is set up by the load-quiz.sh script. 
Here we are making use of the endpoint from the service:



ssh -o strictHostKeyChecking=no node01 /opt/pwx/bin/pxctl cred create --provider s3 --s3-access-key ZZYYXXWWVVUU --s3-secret-key 0ldSup3rS3cr3t --s3-region us-west-2 --s3-endpoint http://$MINIO_ENDPOINT px-s3-store

File: portworx-cloud-credentails/assets/create-objectstores.sh (hunk 4 above)
Comment (vpalazhi) [reply portworx-cloud-credentails/assets/create-objectstores.sh:18]: This script is a template script and it's not actually used by the scenario at all. The mini host is set up by the load-quiz.sh script. 
Here we are making use of the endpoint from the service:



ssh -o strictHostKeyChecking=no  node01 /opt/pwx/bin/pxctl cred create --provider s3 --s3-access-key ZZYYXXWWVVUU --s3-secret-key 0ldSup3rS3cr3t --s3-region us-east-1 --s3-endpoint http://$MINIO_ENDPOINT objectstore-2

File: portworx-cloud-snapshots/assets/create-objectstores.sh (hunk 6 above)
Comment (vpalazhi) [reply portworx-cloud-snapshots/assets/create-objectstores.sh:18]: not used here either. Just a template script.

File: portworx-cloud-snapshots/create-objectstores.sh (hunk 7 above)
Comment (vpalazhi) [reply portworx-cloud-snapshots/create-objectstores.sh:18]: not used by the scenario.

File: portworx-cloud-snapshots/step3.md (hunk 8 above)
Comment (vpalazhi) [reply portworx-cloud-snapshots/step3.md:3]: fixed

File: portworx-k8-sched-snap/assets/create-vols.sh (hunk 10 above)
Comment (vpalazhi) [reply portworx-k8-sched-snap/assets/create-vols.sh:5]: This is fixed already -->
vijin@kodekloud:~/portworx/katacoda-scenarios/portworx-k8-sched-snap$ cat create-vols.sh | grep repl
ssh -o strictHostKeyChecking=no node03 "pxctl volume create ha_volume --nodes=LocalNode --repl=3"
vijin@kodekloud:~/portworx/katacoda-scenarios/portworx-k8-sched-snap$

File: portworx-recap/step0.md (hunk 12 above)
Comment (vpalazhi) [reply portworx-recap/step0.md:10]: Fixed

File: portworx-cloud-credentails/create-objectstores.sh (hunk 5 above)
Comment (vpalazhi) [reply portworx-cloud-credentails/create-objectstores.sh:18]: this file is not used either, minio host created by assets/load-quiz.sh script instead

File: portworx-encryption/assets/create-vols.sh (hunk 9 above)
Comment (vpalazhi) [reply portworx-encryption/assets/create-vols.sh:5]: this is already fixed. the volume is created when the scenario loads.

File: portworx-k8-snapshots/index.json (hunk 11 above)
Comment (vpalazhi) [reply portworx-k8-snapshots/index.json:3]: fixed

File: portworx-resize-pvc/index.json (hunk 13 above)
Comment (vpalazhi) [reply portworx-resize-pvc/index.json:3]: fixed

File: portworx-resize-pvc/step5.md (hunk 14 above)
Comment (vpalazhi) [reply portworx-resize-pvc/step5.md:1]: This step is not used for this scenario (not used in the index.json ) This was copied from another scenario as a template. 

I wil lremove it.

File: portworx-security-roles-contexts/assets/create-vols.sh (hunk 15 above)
Comment (vpalazhi) [reply portworx-security-roles-contexts/assets/create-vols.sh:5]: fixed

File: portworx-snapshots/assets/create-vols.sh (hunk 16 above)
Comment (vpalazhi) [reply portworx-snapshots/assets/create-vols.sh:5]: fixed

File: portworx-troubleshooting/step9.md (hunk 17 above)
Comment (vpalazhi) [reply portworx-troubleshooting/step9.md:3]: fixed

**Pull request merged**